### ISSUE TITLE
AutoPropertyRule.Plain breaks accessing to such properties via interface

### ISSUE THREAD

**Issue by danieldegtyarev** [bridgedotnet/Bridge#3224]
See sample below

### Steps To Reproduce

https://deck.net/9edc43a283d79fc8a73c069389a6b330

```csharp
[Rules(AutoProperty = AutoPropertyRule.Plain)]
public interface IFoo
{
    int Value { get; }
}

public class Foo
{
    int Value { get; }
    
    public Foo()
    {
        Value = 5;
    }
}

public class Program
{
    public static void Main()
    {
        var foo = (IFoo)new Foo();
        
        Console.WriteLine(foo.Value);
    }
}
```

**Comment by geoffreymcgill** [comment]
Hi @danieldegtyarev – Thanks for the report. We will investigate and see if we can get this working correctly.

**Comment by geoffreymcgill** [comment]
The code sample you provided is not valid C#. The Exception message being thrown by Bridge does not match up with the native Exception, but either way, the sample does not compile in native C#.

**Comment by geoffreymcgill** [comment]
I'm not sure exactly what your original intent was, but if I modify so *Foo* implements *IFoo*, then the sample appears to compile properly and outputs the expected result. 

https://deck.net/a0ef44df8d213fab58950685f3530eea

```csharp
[Rules(AutoProperty = AutoPropertyRule.Plain)]
public interface IFoo
{
    int Value { get; }
}

public class Foo : IFoo
{
	int val;
	
	public int Value { get { return this.val; } }
    
    public Foo()
    {
        val = 5;
    }
}

public class Program
{
    public static void Main()
    {
        var foo = (IFoo)new Foo();
        
        Console.WriteLine(foo.Value);
    }
}
```

**Comment by danieldegtyarev** [comment]
Yes, Foo needs implement IFoo, I missed it. 
It fails if in bridge.json 

```
 // Rules to manage generated JavaScript syntax.
  // Default is "Managed"
  "rules": {
    "anonymousType": "Managed",
    "arrayIndex": "Managed",
    "autoProperty": "Plain", // (!)
    "boxing": "Managed",
    "integer": "Managed",
    "lambda": "Managed"
  }```

I dont know how it can be reproduced with Deck.net
Actually I updated my project to the latest version and a new bridge.json had defaults with "autoProperty": "Plain", after that stopped to work with runtime error. I started to investigate and discovered that "autoProperty": "Plain" leads to failure in accessing to autopropety in case of accessing to them via interface.. In my case, I just set everything to managed and it works as expected.
